src/algebra/lie/solvable.lean

```diff
@@ -193,20 +197,46 @@ begin
   exact ⟨⟨k+l, lie_ideal.derived_series_add_eq_bot hk hl⟩⟩,
 end
 
+end lie_algebra
+
 variables {R L}
 
-lemma is_solvable_of_injective {L' : Type w} [lie_ring L'] [lie_algebra R L']
-  [h₁ : is_solvable R L] {f : L' →ₗ⁅R⁆ L} (h₂ : function.injective f) : is_solvable R L' :=
+namespace function
+
+open lie_algebra
+
+lemma injective.lie_algebra_is_solvable [h₁ : is_solvable R L] (h₂ : injective f) :
+  is_solvable R L' :=
 begin
   tactic.unfreeze_local_instances, obtain ⟨k, hk⟩ := h₁,
   use k,
   apply lie_ideal.bot_of_map_eq_bot h₂, rw [eq_bot_iff, ← hk],
-  apply lie_ideal.derived_series_map_le_derived_series,
+  apply lie_ideal.derived_series_map_le,
+end
+
+lemma surjective.lie_algebra_is_solvable [h₁ : is_solvable R L'] (h₂ : surjective f) :
```

Comment:
Does it make sense to also prove lemmas like about the image of a solvable subalgebra? I can imagine that might be useful in practice, because for many proofs you want to work with solvable subalgebras of (nonsolvable) Lie algebras, and sometime you argue that some subalgebra is solvable because it is the image of something else under some map.

Reply:
Sounds like a good idea! I just pushed a commit that aims to do this.